### DISCUSSION TITLE
Design for a simple landing page for Trilium Next

### DISCUSSION THREAD

**Discussion by eliandoran** [TriliumNext/Notes#209] [General]
One of the things I am not very fond of is the fact that Trilium Notes (and implicitly TriliumNext Notes) does not have a landing page of any kind. The only way to find information or to download it is via GitHub. This does not make it very user-friendly.

@TriliumNext/everyone , do we have any designers willing to provide a potential design for the website?

For the moment we need something really simple:

- A big hero/landing section with a screenshot and buttons to download for various platforms.
- A few sections presenting the core features of Trilium Notes.
- A link to the repository.

I can offer to do the development and continuous deployment. @chesspro13 will help us with the domain name.

**Comment by perfectra1n** [comment]
I own the domain `triliumnext.com`, so let me know if you want me to point it somewhere else, do something with it, or transfer it somewhere else! I can also host the site on my Kubernetes cluster behind Cloudflare, if you would like.

**Reply by eliandoran** [reply]
@perfectra1n , regarding hosting I think GitHub Pages is also a good variant, since it allows custom domains as far as I know and I don't expect much traffic for the time being.

**Reply by lokeshsk** [reply]
@eliandoran Do you have any wireframe or prototype? As I can also work on development of the landing page.

**Reply by perfectra1n** [reply]
> @perfectra1n , regarding hosting I think GitHub Pages is also a good variant, since it allows custom domains as far as I know and I don't expect much traffic for the time being.

Oh yeah, duhhh, forgot about GitHub Pages 😂

Just let me know if you want the domain/subdomain to point somewhere else :)

**Reply by rijulraju** [reply]
+1 for GH Pages, it's a great option.

There are a ton of free Jekyll themes out there that'll work out of the box, and require minimal customization: https://jekyllthemes.io/free

Examples:
- https://jekyllthemes.io/theme/hyde
- https://jekyllthemes.io/theme/alembic

**Reply by perfectra1n** [reply]
Yeah I believe that Hugo / Jekyll would be a good choice, I have more familiarity with Hugo.

**Reply by eliandoran** [reply]
> @eliandoran Do you have any wireframe or prototype? As I can also work on development of the landing page.

Unfortunately no, that's why I was hoping to find some web designer to help us out.

**Reply by eliandoran** [reply]
> +1 for GH Pages, it's a great option.
> 
> There are a ton of free Jekyll themes out there that'll work out of the box, and require minimal customization: https://jekyllthemes.io/free
> 
> Examples:
> 
>     * https://jekyllthemes.io/theme/hyde
> 
>     * https://jekyllthemes.io/theme/alembic

@rijulraju , sounds interesting, haven't worked with it.

However I would go against using a pre-made theme and go for something simple and tailor-made for Trilium.
I believe a website must show the personality of an application, and a predefined theme fails to meet that criteria.

**Reply by perfectra1n** [reply]
Would you want to host it from a Trilium instance, and use a theme like https://trilium.rocks/ ?

**Reply by rijulraju** [reply]
@eliandoran That's fair. Happy to help with custom html/css/js.

Jekyll is a static site generator. GitHub Pages supports it natively:
https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/about-github-pages-and-jekyll

GitHub handles generating the site and deploying it, all we would need to do is commit and push changes to the source.

**Reply by eliandoran** [reply]
> Would you want to host it from a Trilium instance, and use a theme like https://trilium.rocks/ ?

That's actually a pretty good idea. @rauenzi , what do you think about that?

**Reply by perfectra1n** [reply]
> > Would you want to host it from a Trilium instance, and use a theme like https://trilium.rocks/ ?
> 
> That's actually a pretty good idea. @rauenzi , what do you think about that?

LOL they might be biased, since it's their site 😂

**Reply by eliandoran** [reply]
@perfectra1n , that's exactly why I asked. 😈

**Reply by meichthys** [reply]
This is also the reason why we have been keeping the docs in markdown format 👍

**Reply by root-hal9000** [reply]
> This is also the reason why we have been keeping the docs in markdown format 👍

So would the docs be included in this page? Is there a way to have the automatic building of docs when merging to main in the Docs repo be useful here? If using GH pages anyway, maybe we just put the website in the docs repo and it can be published that way.


I know we discussed hosting docs in a Trilium instance at the meeting, but that's for the future...

**Reply by zerebos** [reply]
@eliandoran of course I'm very open to it, my share template and everything is open source and I can also spin up a trilium instance specifically for this group

@root-hal9000 it's possible to set up a GitHub action that automatically takes the markdown stored in the repo and push it to a trilium instance. Would just take some work to get setup.

**Comment by meichthys** [comment]
I'm converting this to an Issue since this is definitely something we need. Let's track the progress there: #339
